### ISSUE TITLE
CLI should ask before creating cloud data

### ISSUE THREAD

**Issue by ArmorDarks** [aws-amplify/amplify-cli#576]
**Is your feature request related to a problem? Please describe.**
When you do `amplify init` _it is unclear_ that Amplify will create anything in the cloud`

**Describe the solution you'd like**
Before creating anything during `init` phase, CLI should explicitly ask permission to create anything in the cloud

**Comment by UnleashedMind** [comment]
Thanks for the feedbacks. 
What happened behind the scenes for `amplify init` is explained in the cli document, check [here](https://aws-amplify.github.io/docs/cli/init#amplify-init).
We will have a discussion on this within the team, it seems too much to explain it on the console when running the cli though.

**Comment by ArmorDarks** [comment]
I know, however when I was running it for the first time I didn't expect it anything to deploy _automatically_. That time I regretted I had admin access...

It would be better just to change how `init` works. Instead of deploying configuration of the cloud right away, `init` instead should prepare everything and say that whenever you're ready to configure, run `deploy`. If the cloud wasn't configurated before, `deploy` should do this.

It will be much more predictable and less destructive behaviour.

**Comment by jkeys-ecg-nmsu** [comment]
Agreed that `amplify init` probably shouldn't provision AWS resources until the first `amplify push`.

**Comment by HknLof** [comment]
This feature also makes sense in AWS Accounts which are governed under AWS orgs.
Scenario:

3 Accounts: dev, staging, prod

Only for dev one has "human" assumable role and the others are staged through CICD. Let's say we create an env dev in dev and just want to stage the same setup all the way towards prod.

Then `aws init` actually fails as it does not have local state for the environment staging and prod.
